app/src/main/java/com/example/pygmyhippo/organizer/EventFragment.java

```diff
@@ -0,0 +1,213 @@
+        });
+
+        // Set up the listener for viewing entrants button
+        Button viewEntrantsButton = view.findViewById(R.id.button_view_entrants);
+        viewEntrantsButton.setOnClickListener(view1 -> {
+            Navigation.findNavController(view1).navigate(R.id.action_event_fragment_to_view_entrants_fragment);
+        });
+
+        TextView eventNameView = view.findViewById(R.id.u_eventNameView);
+        TextView eventDateView = view.findViewById(R.id.u_eventDateView);
+        TextView eventTimeView = view.findViewById(R.id.u_eventTimeView);
+        TextView eventOrganizerView = view.findViewById(R.id.u_organizerNameView);
+        TextView eventLocationView = view.findViewById(R.id.u_eventLocationView);
+        TextView eventCostView = view.findViewById(R.id.u_eventCostView);
+        TextView eventAboutDescriptionView = view.findViewById(R.id.u_aboutEventDescriptionView);
+        Button closeEventButton = view.findViewById(R.id.close_event_button);
+
+        // set with hardcoded values, set the view in the same wy form the event received from the constructor
+        // TODO: database event info to be done here
+        event = hardcodeEvent();
+        event.setEventWinnersCount(5);
+
+        eventNameView.setText(event.getEventTitle());
+        eventDateView.setText(event.getDate());
+        eventTimeView.setText(event.getTime());
+        eventOrganizerView.setText(event.getOrganiserID());
+        eventLocationView.setText(event.getLocation());
+        eventCostView.setText(event.getCost());
+        eventAboutDescriptionView.setText(event.getDescription());
+
+        // If the event is closed, then change the style of the draw button
+        if (event.getEventStatus().value.equals("cancelled")) {
+            closeEventButton.setBackgroundColor(0xFFA4A8C3);
+            closeEventButton.setText("Lottery closed");
+            closeEventButton.setTextColor(0xFF3A5983);
+            closeEventButton.setTextSize(20);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.organiser_fragment_event, container, false);
+        Button closeEventButton = view.findViewById(R.id.close_event_button);
+
+        closeEventButton.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view) {
+                // Only do the draw if the event hasn't already been closed
+                if (event.getEventStatus().value.equals("ongoing") && !event.getEntrants().isEmpty()) {
+                    // Change the view of the button
+                    closeEventButton.setBackgroundColor(0xFFA4A8C3);
+                    closeEventButton.setText("Lottery closed");
+                    closeEventButton.setClickable(false);
```

Comment:
just wondering, the organizers can't reopen an event right?

Reply:
No. I don't think we need that functionality either. Unless you recall a scenario or story where we do